services/workers/src/test/java/org/sagebionetworks/table/worker/TableViewIntegrationTest.java

```diff
@@ -1575,6 +1574,45 @@ public void testEntityView_multipleValueColumnRoundTrip_changeMaxListLength() th
 		).getMessage();
 		assertEquals("maximumListLength for ColumnModel \"stringList\" must be at least: 3", error);
 	}
+
+
+	@Test
+	public void testEntityView_multipleValueColumnRoundTrip_updateRowGreaterThanMaxListLength() throws Exception {
+		defaultColumnIds.add(stringListColumn.getId());
+		createFileView();
+
+		assertTrue(fileCount >= 2, "setup() needs to create at least 2 entities for this test to work");
+
+		Long fileId = KeyFactory.stringToKey(fileIds.get(0));
+
+		//set annotations for 2 files
+		Annotations fileAnnotation1 = entityManager.getAnnotations(adminUserInfo, fileIds.get(0));
+		AnnotationsV2TestUtils.putAnnotations(fileAnnotation1, stringListColumn.getName(), Arrays.asList("val1", "val2", "val3"), AnnotationsValueType.STRING);
+		entityManager.updateAnnotations(adminUserInfo, fileIds.get(0), fileAnnotation1);
+
+		Annotations fileAnnotation2 = entityManager.getAnnotations(adminUserInfo, fileIds.get(1));
+		AnnotationsV2TestUtils.putAnnotations(fileAnnotation2, stringListColumn.getName(), Arrays.asList("val2", "val3"), AnnotationsValueType.STRING);
+		entityManager.updateAnnotations(adminUserInfo, fileIds.get(1), fileAnnotation2);
+
+
+		waitForEntityReplication(fileViewId, fileIds.get(0));
+
+
+		QueryResultBundle result = waitForConsistentQuery(adminUserInfo, "select " + stringListColumn.getName() + " from " + fileViewId, fileCount);
+		assertEquals(fileCount, result.getQueryCount());
+
+		//update annotations via row changes to exceed maximumListLength
+		RowSet rowset = result.getQueryResult().getQueryResults();
+		rowset.getRows().get(0).setValues(Collections.singletonList("[\"val1\",\"val2\",\"val3\",\"val4\"]"));
+
+		// wait for the change to complete
+//		startAndWaitForJob(adminUserInfo, transaction, TableUpdateTransactionResponse.class);
+
+		String error = assertThrows(AsynchJobFailedException.class, () ->
+				updateView(rowset,fileViewId)
+		).getMessage();
+		assertEquals("Value at [0,16] was not a valid STRING_LIST. Exceeds the maximum number of list elements defined in the ColumnModel (3): \"[\"val1\",\"val2\",\"val3\",\"val4\"]\"", error);

```

Comment:
Can you double check if we truncate the number of values at the end of the message? The error message might become extremely long, maybe we can remove the list of values in the error message (e.g. we might just need the total count)?